randovania/games/super_metroid/patcher/super_duper_metroid_patcher.py

```diff
@@ -54,24 +57,125 @@ def sm_pickup_details_to_patcher(detail: pickup_exporter.ExportedPickupDetails,
         "Power Bombs": "Power Bomb Expansion",
 
-        "Nothing": "Missile Expansion",
+        "Nothing": "No Item",
+    }
+
+    _effect = {
+        "Energy Tank": "Get Energy Tank",
+        "Missile Expansion": "Get Missile Expansion",
+        "Super Missile Expansion": "Get Super Missile Expansion",
+        "Power Bomb Expansion": "Get Power Bomb Expansion",
+        "Grapple Beam": "Get Grapple Beam",
+        "X-Ray Scope": "Get X-Ray Scope",
+        "Varia Suit": "Get Varia Suit",
+        "Spring Ball": "Get Spring Ball",
+        "Morph Ball": "Get Morph Ball",
+        "Screw Attack": "Get Screw Attack",
+        "Hi-Jump Boots": "Get Hi-Jump Boots",
+        "Space Jump": "Get Space Jump",
+        "Speed Booster": "Get Speed Booster",
+        "Charge Beam": "Get Charge Beam",
+        "Ice Beam": "Get Ice Beam",
+        "Wave Beam": "Get Wave Beam",
+        "Spazer Beam": "Get Spazer Beam",
+        "Plasma Beam": "Get Plasma Beam",
+        "Morph Ball Bombs": "Get Morph Ball Bombs",
+        "Reserve Tank": "Get Reserve Tank",
+        "Gravity Suit": "Get Gravity Suit",
+        "No Item": "No Effect",
     }
 
+    item_name = _mapping.get(pickup_type, pickup_type)
+
     result = {
-        "item_name": _mapping.get(pickup_type, pickup_type),
-        "quantity": count,
+        "item_name": item_name,
+        "quantity_given": count,
+        "pickup_effect": _effect[item_name],
         "pickup_index": detail.index.index,
+        "native_sprite_name": item_name,
+        "owner_name": None,
+    }
+
+    return result
+
+
+def sm_starting_items_to_patcher(item: ItemResourceInfo, quantity: int) -> dict:
+    _mapping = {
+        "Missile": "Missile Expansion",
+        "Super Missile": "Super Missile Expansion",
+        "Power Bombs": "Power Bomb Expansion",
+
+        "Nothing": "No Item",
     }
 
+    _effect = {
+        "Energy Tank": "Get Energy Tank",
+        "Missile Expansion": "Get Missile Expansion",
+        "Super Missile Expansion": "Get Super Missile Expansion",
+        "Power Bomb Expansion": "Get Power Bomb Expansion",
+        "Grapple Beam": "Get Grapple Beam",
+        "X-Ray Scope": "Get X-Ray Scope",
+        "Varia Suit": "Get Varia Suit",
+        "Spring Ball": "Get Spring Ball",
+        "Morph Ball": "Get Morph Ball",
+        "Screw Attack": "Get Screw Attack",
+        "Hi-Jump Boots": "Get Hi-Jump Boots",
+        "Space Jump": "Get Space Jump",
+        "Speed Booster": "Get Speed Booster",
+        "Charge Beam": "Get Charge Beam",
+        "Ice Beam": "Get Ice Beam",
+        "Wave Beam": "Get Wave Beam",
+        "Spazer Beam": "Get Spazer Beam",
+        "Plasma Beam": "Get Plasma Beam",
+        "Morph Ball Bombs": "Get Morph Ball Bombs",
+        "Reserve Tank": "Get Reserve Tank",
+        "Gravity Suit": "Get Gravity Suit",
+        "No Item": "No Effect",
+    }
+
+    item_name = _mapping.get(item.long_name, item.long_name)
```

Comment:
if these lists are identical between the two functions, you should avoid defining them twice

Reply:
actually, better yet - I recommend putting this data in the `extra` field in the item database, instead of list like this